test/jdk/tools/jar/ReproducibleJar.java

```diff
@@ -0,0 +1,296 @@
+        Assert.assertTrue(dirInner.mkdir());
+        try (PrintWriter pw = new PrintWriter(fileInner)) {
+            pw.println("hello, world");
+        }
+
+        Assert.assertTrue(dirOuter.exists());
+        Assert.assertTrue(dirInner.exists());
+        Assert.assertTrue(fileInner.exists());
+    }
+
+    static void checkFileTime(long now, long original) throws Throwable {
+        if (isTimeSettingChanged()) {
+            return;
+        }
+
+        if (Math.abs(now - original) > PRECISION) {
+            // If original time is after UNIX 2038 32bit rollover
+            // and the now time is exactly the rollover time, then assume
+            // running on a file system that only supports to 2038 (e.g.XFS) and pass test
+            if (FileTime.fromMillis(original).toInstant().isAfter(unix2038Rollover) &&
+                FileTime.fromMillis(now).toInstant().equals(unix2038Rollover)) {
+                System.out.println("Checking file time after Unix 2038 rollover," +
+                                   " and extracted file time is " + unix2038RolloverTime + ", " +
+                                   " Assuming restricted file system, pass file time check.");
+            } else {
+                throw new AssertionError("checkFileTime failed," +
+                                         " extracted to " +  FileTime.fromMillis(now) +
+                                         ", expected to be close to " + FileTime.fromMillis(original));
+            }
+        }
+    }
+
+    static void checkSameContent(File f1, File f2) throws Throwable {
+        byte[] ba1 = Files.readAllBytes(f1.toPath());
+        byte[] ba2 = Files.readAllBytes(f2.toPath());
+        if (!Arrays.equals(ba1, ba2)) {
+            throw new AssertionError("jar content differs:" + f1 + " != " + f2);
+        }
+    }
+
+    private static boolean isTimeSettingChanged() {
+        TimeZone currentTZ = TimeZone.getDefault();
+        boolean currentDST = currentTZ.inDaylightTime(new Date());
+        if (!currentTZ.equals(TZ) || currentDST != DST) {
+            System.out.println("Timezone or DST has changed during ReproducibleJar testcase execution. Test skipped");
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private static boolean isInTransition() {
+        boolean inTransition = false;
+
+        var date = new Date();
+        var defZone = ZoneId.systemDefault();
+        if (defZone.getRules().getTransition(
+                date.toInstant().atZone(defZone).toLocalDateTime()) != null) {
+            System.out.println("ReproducibleJar testcase being run during Zone offset transition.  Test skipped.");
+            inTransition = true;
+        }
+
+        return inTransition;
+    }
+
+    static boolean cleanup(File dir) throws Throwable {
+        boolean rc = true;
+        File[] x = dir.listFiles();
+        if (x != null) {
+            for (int i = 0; i < x.length; i++) {
```

Comment:
Could use enhanced for loop here if you desire

Reply:
done